### ISSUE TITLE
Add fails when model inputs are passed to it

### ISSUE THREAD

**Issue by soumyac1999** [flexflow/FlexFlow#386]
File to reproduce (a simple modification to https://github.com/flexflow/FlexFlow/blob/master/examples/python/keras/unary.py):
```python
from flexflow.keras.models import Model, Sequential
from flexflow.keras.layers import Add, Subtract, Flatten, Dense, Activation, Conv2D, MaxPooling2D, Concatenate, add, subtract, Input
import flexflow.keras.optimizers

import flexflow.core as ff
import numpy as np

def top_level_task():
  input1 = Input(shape=(16, ), dtype="float32")
  x1 = Dense(8, activation='relu')(input1)
  input2 = Input(shape=(8,), dtype="float32")
  added = Add()([x1, input2])

  out = Dense(4)(added)
  model = keras.models.Model([input1, input2], out)

  opt = flexflow.keras.optimizers.SGD(learning_rate=0.01)
  model.compile(optimizer=opt, loss='mean_squared_error', metrics=['mean_squared_error'])
  print(model.summary())
  model.fit(
    x = [np.random.randn(300, 16).astype(np.float32), np.random.randn(300, 8).astype(np.float32)],
    y = np.random.randn(300, 4).astype(np.float32)
  )

if __name__ == '__main__':
  top_level_task()
```

Error Message:
```
flexflow_python: /home/soumya/flexflow/deps/legion/runtime/realm/dynamic_templates.inl:153: static void Realm::DynamicTemplates::IntList<MIN, MAX>::demux(int, T1, T2) [with TARGET = Realm::DynamicTemplates::ListProduct2<Realm::DynamicTemplates::IntList<1, 5>, Realm::DynamicTemplates::TypeListElem<int, Realm::DynamicTemplates::TypeListElem<unsigned int, Realm::DynamicTemplates::TypeListElem<long long int, Realm::DynamicTemplates::TypeListTerm> > > >::DemuxHelper1<Legion::Internal::NT_TemplateHelper::DimHelper>; T1 = unsigned int; T2 = int*; int MIN = 1; int MAX = 5]: Assertion `(MIN <= index) && (index <= MAX)' failed.
```

**Comment by lockshaw** [comment]
Looks like a failure in Legion/Realm, so @jiazhihao should probably take a look

**Comment by jiazhihao** [comment]
This should be fixed by PR #391
